src/Illuminate/Database/Schema/PostgresSchemaState.php

```diff
@@ -51,7 +51,13 @@ protected function appendMigrationData(string $path)
      */
     public function load($path)
     {
-        $process = $this->makeProcess('PGPASSWORD=$LARAVEL_LOAD_PASSWORD pg_restore --host=$LARAVEL_LOAD_HOST --port=$LARAVEL_LOAD_PORT --username=$LARAVEL_LOAD_USER --dbname=$LARAVEL_LOAD_DATABASE $LARAVEL_LOAD_PATH');
+        $command = 'PGPASSWORD=$LARAVEL_LOAD_PASSWORD pg_restore --host=$LARAVEL_LOAD_HOST --port=$LARAVEL_LOAD_PORT --username=$LARAVEL_LOAD_USER --dbname=$LARAVEL_LOAD_DATABASE $LARAVEL_LOAD_PATH';
+
+        if (preg_match('/\.sql$/', $path) !== false) {
```

Comment:
I believe this line is wrong. When I'm trying to dump and do a fresh migrate now I get an error due to the fact that it's trying to load a .dump file using `psql` instead of `pg_restore`.

I believe `if (fnmatch('*.sql', $path))` might be an easier option here. `preg_match` does not give a boolean result, it [returns an int value](https://www.php.net/preg_match).